### PR TITLE
Enhance workflow scripts

### PR DESCRIPTION
**Pull request by ghutchis**
Several key enhancements:
- workflows can have no options (e.g., just flatten already)
- workflows can return data in multiple formats (e.g., xyz)
- fix GUI bug where the dialog would continually show old dialogs

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ghutchis** [comment]
More example workflows are here:
https://github.com/OpenChemistry/avogadro-ugm2018/blob/master/workflows/

For example, `pip install scikit-nano` and then run:
https://github.com/OpenChemistry/avogadro-ugm2018/blob/master/workflows/nanotube.py

screen shot 2018-07-28 at 12 44 27 am

File: avogadro/qtplugins/workflows/workflow.cpp [hunk 1]
```diff
@@ -238,9 +247,9 @@ void Workflow::updateActions()
 {
   m_actions.clear();
 
-  QAction* action = new QAction(tr("Set Python Path..."), this);
-  connect(action, SIGNAL(triggered()), SLOT(configurePython()));
-  m_actions << action;
+  //  QAction* action = new QAction(tr("Set Python Path..."), this);
```
[the remark this answers is not in the document]
Comment (ghutchis) [reply avogadro/qtplugins/workflows/workflow.cpp]: This is a duplicate. There's one for the generators and this is an "extra" one.

The best solution (not in this pull request) is to set this through the `avogadroapp` or something more global.

**Comment by ghutchis** [comment]
@cryos - I'd definitely like to merge this soon - it's a good working state to allow others to add useful scripts and it seems the main problem was confusion over a duplicate menu item for configuring Python...

**Comment by cryos** [comment]
Sounds good, I have been traveling, and was hoping to test this actually working on a Linux machine but had no real objection other than the concern about removing the Python thing without comment.

**Pull request merged**